core/src/main/java/org/mule/transport/AbstractMessageDispatcher.java

```diff
@@ -287,7 +288,11 @@ public void run()
                     {
                         try
                         {
-                            event.getReplyToHandler().processReplyTo(createResponseEvent(result, event), null, null);
+                            ((ThreadSafeAccess) result).resetAccessControl();
```

Comment:
Variable result is a MuleMessage, you need to check that it is also a ThreadSafeAccess before casting.